stl/inc/algorithm

```diff
@@ -2065,21 +2181,27 @@ namespace ranges {
             _Rng1&& _Range1, _Rng2&& _Range2, _Pr _Pred = {}, _Pj1 _Proj1 = {}, _Pj2 _Proj2 = {}) const {
             if constexpr (random_access_range<_Rng1> && sized_range<_Rng1>
                 && random_access_range<_Rng2> && sized_range<_Rng2>) {
-                return _Random_access_sized_ranges(
-                    _RANGES begin(_Range1), _RANGES distance(_Range1),
-                    _RANGES begin(_Range2), _RANGES distance(_Range2),
+                auto _First1 = _RANGES begin(_Range1);

```

Comment:
Nit in other similar functions you pulled this out if the `if constexpr`

Reply:
Yeah, I think I got distracted here for the 100th time trying to convince myself to do the `if constexpr` inside a lambda so I can factor out the common `_Rewrap_subrange` tail a la:
```c++
auto _First1 = _RANGES begin(_Range1);
auto _Meow = [&] {
    if constexpr (random_access_range<_Rng1> && sized_range<_Rng1>
        && random_access_range<_Rng2> && sized_range<_Rng2>) {
        return _Random_access_sized_ranges(
            _Get_unwrapped(_First1), _RANGES distance(_Range1),
            _Get_unwrapped(_RANGES begin(_Range2)), _RANGES distance(_Range2),
            _Pass_fn(_Pred), _Pass_fn(_Proj1), _Pass_fn(_Proj2));
    } else if constexpr (bidirectional_range<_Rng1> && common_range<_Rng1>
        && bidirectional_range<_Rng2> && common_range<_Rng2>) {
        return _Bidi_common_ranges(
            _Get_unwrapped(_First1), _Get_unwrapped(_RANGES end(_Range1)),
            _Get_unwrapped(_RANGES begin(_Range2)), _Get_unwrapped(_RANGES end(_Range2)),
            _Pass_fn(_Pred), _Pass_fn(_Proj1), _Pass_fn(_Proj2));
    } else {
        return _Forward_ranges(
            _Get_unwrapped(_First1), _Get_unwrapped(_RANGES end(_Range1)),
            _Get_unwrapped(_RANGES begin(_Range2)), _Get_unwrapped(_RANGES end(_Range2)),
            _Pass_fn(_Pred), _Pass_fn(_Proj1), _Pass_fn(_Proj2));
    }
};
return _Rewrap_subrange<borrowed_subrange_t<_Rng1>>(_First1, _Meow());
```
but it feels like a hack. Also it's just occurred to me that I really need `_Begin_unwrapped` and `_End_unwrapped` CPOs. `_Get_unwrapped(_RANGES begin(_Range2))` is quite a mouthful.